core-bundle/src/Resources/contao/pages/PageRegular.php

```diff
@@ -808,6 +811,48 @@ protected function createFooterScripts($objLayout)
 			}
 		}
 
+		// Add search index meta data
+		if ($objPage !== null)
+		{
+			$noSearch = (bool) $objPage->noSearch;
+
+			// Backwards compatibility: Do not use $GLOBALS['TL_NOINDEX_KEYS'] anymore
+			// but make sure your page type delivers the correct meta robots tag
+			foreach (array_keys($_GET) as $key)
+			{
+				foreach ($GLOBALS['TL_NOINDEX_KEYS'] as $noIndexKey)
+				{
+					if (preg_match('/^' . $noIndexKey . '$/', $key))
```

Comment:
I think this would be more efficient if we replaced the inner foreach loop with `preg_grep()`, wouldn't it?

Reply:
How would that work? `preg_grep()` applies the same regex to every element. We need the opposite: We need a regexp for every element to apply on one key :)

Reply:
> `preg_grep()` applies the same regex to every element

Exactly. We could convert TL_NOINDEX_KEYS to a regex and then apply it to `array_keys($_GET)`.

```php
if (preg_grep('/^(id|file|page|page_.*|keywords)$/', array_keys($_GET))) {
    $noSearch = true;
}
```